yggdrasilffi/src/lib.rs

```diff
@@ -41,17 +42,32 @@ impl<T> From<Result<Option<T>, FFIError>> for Response<T> {
             Err(e) => Response {
                 status_code: ResponseCode::Error,
                 value: None,
-                error_message: Some(format!("{:?}", e)),
+                error_message: Some(e.to_string()),
             },
         }
     }
 }
 
-#[derive(Debug)]
 enum FFIError {
     Utf8Error,
-    InvalidJson,
     NullError,
+    InvalidJson(String),
+    PartialUpdate(Vec<EvalWarning>),
+}
+
+impl Display for FFIError {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            FFIError::Utf8Error => write!(f, "Detected a non UTF-8 string in the input, this is a serious issue and you should report his as a bug."),
+            FFIError::NullError => write!(f, "Null error detected, this is a serious issue and you should report this as a bug."),
+            FFIError::InvalidJson(message) => write!(f, "Failed to parse JSON: {}", message),
+            FFIError::PartialUpdate(messages) => write!(
```

Comment:
Bit hard to grok at a glance if you also haven't seen the EvalWarning code but EvalWarning is `Debug` so this becomes a list of 

{
   "toggleName": "some tooggle",
   "message": "Some error message explaining why your feature toggle is broken"  
}

In practice, this only happens if you break both the stategy_upgrade and the strategy_parsing code i.e. you need to introduce a bug that allows Ygg to accept a toggle that it can't compile.

If this does happen, it should only occur in the thread that does the toggle fetching so `isEnabled` shouldn't be affected